### DISCUSSION TITLE
Feature request: clipping path support for block references

### DISCUSSION THREAD

**Discussion by subsurfaceiodev** [mozman/ezdxf#760] [Ideas]
Hi there! It would be really helpful to be able to set clipping boundaries for block reference objects (XCLIP command in AutoCAD). I didn't find any documentation about it. SUPERHATCH in summary inserts multiple references to the same block filling selected boundary and then applies XCLIP to the blocks near the boundary. Thanks!

**Comment by mozman** [comment]
This is a complex feature which requires multiple DXF entities for implementation.

The clipping path is stored in a `SPATIAL_FILTER` object in the `OBJECTS` section and linked to the `INSERT` entity by a chain of `DICTIONARY` entities starting at the extension dictionary of the `INSERT` entity. The `SPATIAL_FILTER` entity is documented in the DXF reference and supported by all DXF versions from R2000 onward. The feature is supported by BricsCAD and I assume also by other CAD applications.

- usable, because not an AutoCAD exclusive feature
- doable, but requires time, especially the copy and transformation support
- not in the near future

**Comment by mozman** [comment]
Clippings paths for block references are supported in v1.2.0. [docs](https://ezdxf.mozman.at/docs/tools/xclip.html)
